src/initialization/InitDistribution.cpp

```diff
@@ -471,9 +476,22 @@ namespace impactx
         }
         else if (track == "envelope")
         {
+            std::string space_charge_model = "2D"; // 2D space charge as existing default
+            pp_algo.query("space_charge_model", space_charge_model);
+            amrex::ParticleReal intensity = 0.0; // bunch charge (C) for 3D model, beam current (A) for 2D model
+
+            if (space_charge_model == "3D") {
+                pp_dist.query("charge", intensity);
+                throw std::runtime_error("3D space charge model not yet implemented in envelope mode.");
+            } else if (space_charge_model == "2D") {
+                pp_dist.query("current", intensity);
+            } else {
+                throw std::runtime_error("Unknown space_charge_model (use '2D' or '3D') ");
+            }
```

Comment:
I think I want to unify this with `algo.space_charge`. There is no need for two parameters, we can just do `algo.space_charge = false` (default), `= 2D` and `= 3D`.

Reply:
This makes sense, although it will break the existing space charge examples, where `algo.space_charge = true`.  In that case, maybe we should push the change through as a follow-up, since it also impacts the logic outside the envelope tracking?

Reply:
I think I can implement it in a backwards compatible way, will give it a try tomorrow.

Reply:
I pushed a backwards compatible change and it currently passes CI. This is what existing users will see (True defaults now to 3D space charge and raises a warning to use the new syntax.)

I will push another commit afterwards that updates all examples to the new syntax.